src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1347,5 +1317,326 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+        {
+            ProcessStartInfo info = new ProcessStartInfo
+            {
+                LoadUserProfile = true,
+                UseCredentialsForNetworkingOnly = true,
+                UserName = "dummy",
+                PasswordInClearText = "not used, because ArgumentException should be thrown before",
+                FileName = "notepad.exe",
+                Arguments = null,
+                WindowStyle = ProcessWindowStyle.Minimized
+            };
+
+            Assert.Throws<ArgumentException>("startInfo", () =>
+            {
+                using (var process = Process.Start(info))
+                {
+                    Assert.False(process != null, $"Process started despite incompatible options {nameof(info.LoadUserProfile)} and {nameof(info.UseCredentialsForNetworkingOnly)} were enabled");
+                }
+            });
+        }
+
+        [ConditionalFact(nameof(IsAdmin_IsNotNano_RemoteExecutorIsSupported))] // Nano has no "netapi32.dll", Admin rights are required
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [OuterLoop("Requires admin privileges")]
+        public void TestUserNetworkCredentialsPropertiesOnWindows()
+        {
+            string testFilePath = GetTestFilePath();
+            Assert.False(string.IsNullOrWhiteSpace(testFilePath), $"Path to test file should not be empty: {testFilePath}");
+
+            string testFilePathRoot = Path.GetPathRoot(testFilePath);
+            const string ShareName = "testForDotNet";
+            using TestFileShare fileShare = new TestFileShare(ShareName, Path.GetDirectoryName(testFilePath));
+            string testFileUncPath = $"\\\\localhost\\{ShareName}\\{Path.GetFileName(testFilePath)}";
+            string testFileContent = Guid.NewGuid().ToString();
+            File.WriteAllText(testFilePath, testFileContent);
+
+            const string LocalPath = nameof(LocalPath);
+            const string UncPath = nameof(UncPath);
+            using Process p = CreateProcess(() =>
+            {
+                // Read file content using network credentials and output it for assertion comparison
+                Console.Write(File.ReadAllText(Environment.GetEnvironmentVariable(UncPath)));
+                Console.Write(ItemSeparator);
+
+                try
+                {
+                    // Read file content using current user - should fail, because impersonated user
+                    // isn't explicitly authorized to access the file.
+                    _ = File.ReadAllText(Environment.GetEnvironmentVariable(LocalPath));
+                    Console.Write('1');
+                }
+                catch (SecurityException)
+                {
+                    Console.Write('0');
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    Console.Write('0');
+                }
+
+                return RemoteExecutor.SuccessExitCode;
```

Comment:
We could simplify it a bit and just use different exit codes to indicate failure rather than writing to standard output:

```suggestion
                try
                {
                    // Read file content using current user - should fail, because impersonated user
                    // isn't explicitly authorized to access the file.
                    _ = File.ReadAllText(Environment.GetEnvironmentVariable(LocalPath));
                    
                    return -1;
                }
                catch (Exception ex) when (ex is SecurityException or UnauthorizedAccessException)
                {
                    return RemoteExecutor.SuccessExitCode;
                }
```